pkg/util/sketch/heap.go

```diff
@@ -0,0 +1,72 @@
+package sketch
+
+import (
+	"container/heap"
+)
+
+type node struct {
+	event string
+	count uint32
+	// used for the container heap Fix function
+	index           uint16
+	sketchPositions []uint32
+}
+
+type MinHeap []*node
```

Comment:
Should we reuse an existing heap implementation?

Reply:
We could but I'd have to look at the ones we have so far to see if we can use it. I don't think it's too important for us to try to dedupe this particular bit of the code.

Reply:
It's not that much but we would avoid bugs. On the other hand we can change and optimize this code as we see fit for our case.

Reply:
the heap is probably a "one and done" implementation anyways, the one I have here is basically straight from the go source examples for using the `container.Heap` package where they implement a priority queue, except reversed slightly so that it's a min heap: https://pkg.go.dev/container/heap#example-package-PriorityQueue